Chore/update database usage (#12602)
supabase/supabase: apps/docs/pages/guides/platform/database-size.mdx

Supabase uses network-attached storage to balance performance with scalability.

### Free tier project

For [**Free tier projects**](#free-tier-project), the database can enter ['read only' mode when you exceed the 500mb limit](#free-tier-project-read-only-mode).

#### Free tier project 'read-only' mode

Free tier projects can enter read-only mode when exceeding the 500mb database size limit.

In read-only mode, clients will encounter errors such as `cannot execute INSERT in a read-only transaction`.
Regular operation (read-write mode) is automatically re-enabled once database storage is below the 500mb database size limit.

#### Increasing available database space

> If your project is on the free tier, there is a 500mb Database size limit.

Reviewer: ```suggestion
The Free Tier has a 500mb Database size limit.
```
